src/metabase/driver/common/parameters/values.clj

```diff
@@ -265,17 +265,18 @@
 (s/defn ^:private update-filter-for-field-type :- ParsedParamValue
   "Update a Field Filter with a textual, or sequence of textual, values. The base type and semantic type of the field
   are used to determine what 'semantic' type interpretation is required (e.g. for UUID fields)."
-  [{{effective_type :effective_type, :as _field} :field, {value :value} :value, :as field-filter} :- FieldFilter]
-  (let [new-value (cond
+  [{field :field, {value :value} :value, :as field-filter} :- FieldFilter]
+  (let [effective-type (or (:effective_type field) (:base_type field))
```

Comment:
I thought Field was always supposed to have `effective_type`? Any idea why it would be missing here?

If we're determining it from the `:base-type` information in the field ref, maybe we should also add `:effective_type` at the same place we do that

Reply:
I think my thinking was that our inferencing code doesn't use effective_type yet. So Fields capitol F always have an effective type. That's not necessarily the case with aggregations and the like.

Reply:
Trying to move to a world where everyone only considers effective type, and only the query processor's really need to know about base type when emitting queries but that change was a bit too big especially when you had a really big field ref refactor at the same time.

Reply:
Yeah. That wasn't great timing on my part. Sorry about that. I think this change makes sense for now

Reply:
it wasn't just yours. the rename from special_type to semantic_type touched 140 or so files. it was a big change where basically nothing was supposed to happen. and adding inferencing code into an already huge change made it impossible. but there was a lot of upheaval outside of it too and it just was far better to delay it.